### PR TITLE
Add the ability to define maven repositories when creating a project

### LINKED ISSUE #9588: code.quarkus.io should have option for including maven repository in its resulting projects
for code.quarkus.redhat.com there are the issue that some of the extensions are only served from maven.repository.redhat.com.  You can have similar for other platforms or company specific code.quarku instances.

proposal is that code.quarkus.io gets a flag or env setting that enables it to include a maven repository setup for its generated projects.

### PR DESCRIPTION
**Pull request by ia3andy**
Fixes #9588

this is only for Maven project

### CODE REVIEW

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/codegen/rest/BasicRestProjectGenerator.java [hunk 1]
```diff
@@ -95,6 +95,12 @@ private boolean initBuildTool() throws IOException {
             if (!invocation.hasValue(BUILD_DIRECTORY)) {
                 invocation.setValue(BUILD_DIRECTORY, buildTool.getBuildDirectory());
             }
+            if (!invocation.hasValue(MAVEN_REPOSITORIES)) {
+                invocation.setValue(MAVEN_REPOSITORIES, "\n");
```
Comment (gastaldi) [inline independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/codegen/rest/BasicRestProjectGenerator.java:99]: This is odd, a newline?

**Review by gastaldi — APPROVED** [review]
LGTM

File: devtools/platform-descriptor-json/src/main/resources/templates/basic-rest/java/build.gradle-template.ftl [hunk 2]
```diff
@@ -4,7 +4,7 @@ plugins {
 }
 
 repositories {
-     mavenLocal()
+     mavenLocal()${maven_repositories}
```
Comment (maxandersen) [inline devtools/platform-descriptor-json/src/main/resources/templates/basic-rest/java/build.gradle-template.ftl:7]: ordering matters - mavencentral should come before red hat maven to not get super slow for artifacts that are from mavencentral.

Comment (ia3andy) [reply devtools/platform-descriptor-json/src/main/resources/templates/basic-rest/java/build.gradle-template.ftl:7]: @maxandersen already merged ^^ I'll push a fix

**Pull request merged**